### PR TITLE
Create decorator

### PR DESCRIPTION
**Pull request by mohashyne**
# New  Features Added

   
- [ ]  Implemented Decorator pattern in Ruby.
- [ ] Created Nameable class with correct_name method raising NotImplementedError.
- [ ] Modified Person class to inherit from Nameable and implement correct_name.
- [ ] Prepared base Decorator class inheriting from Nameable, with correct_name using @nameable.
- [ ] Created CapitalizeDecorator class inheriting from Decorator, capitalizing @nameable.correct_name.
- [ ]  Created TrimmerDecorator class inheriting from Decorator, limiting @nameable.correct_name to max 10 characters.
- [ ] Tested decorators with Person, CapitalizeDecorator, and TrimmerDecorator.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require_relative 'Decorator'
```
Comment (anagudelogu) [inline trimmer_decorator.rb]: - [ ] Please consider fixing the casing in your `require_relative`. The name of the file you're importing is `decorator`, not `Decorator`. 🤓 🔧

File: capitalize_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Decorator'
```
Comment (anagudelogu) [inline capitalize_decorator.rb]: - [ ] Please consider fixing the casing in your `require_relative`. The name of the file you're importing is `decorator`, not `Decorator`. 🤓 🔧

File: person.rb [hunk 3]
```diff
@@ -1,9 +1,19 @@
-class Person
-  def initialize(name: 'Unknown', age: nil, parent_permission: true)
+require_relative 'nameable'
+require_relative 'Decorator'
+require_relative 'Trimmer_Decorator'
+require_relative 'Capitalize_Decorator'
```
Comment (anagudelogu) [inline person.rb]: - [ ] **[OPTIONAL]** These imports are wrong too, however, as they are imported for testing purposes (logging the results into the console) I suggest that once you test the code, you remove them to keep your class files clean at all times. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @mohashyne, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @mohashyne 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**